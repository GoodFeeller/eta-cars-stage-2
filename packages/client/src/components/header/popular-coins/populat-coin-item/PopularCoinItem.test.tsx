import { ICoin } from '../../../../types/ICoin'
import { render, screen } from '@testing-library/react'
import PopularCoinItem from './PopularCoinItem'
import { BrowserRouter, useNavigate } from 'react-router-dom'
import userEvent from '@testing-library/user-event'

const coin: ICoin = {
  rank: '1',
  symbol: 'BTC',
  name: 'Bitcoin',
  id: 'bitcoin',
  priceUsd: '21000.2313125256326',
  maxSupply: '1000000.223',
  supply: '141312',
  marketCapUsd: '32131412123.231242412321',
  changePercent24Hr: '0.24',
}

describe('PopularCoinItem test', () => {
  test('PopularCoinItem snapshot test', () => {
    render(
      <BrowserRouter>
        <PopularCoinItem coin={coin} />
      </BrowserRouter>,
    )
    const coinItem = screen.getByTestId('coin-item')
    expect(coinItem).toMatchSnapshot()
  })

  test('Img must have src assets/icons/btc@2x.png', () => {
    render(
      <BrowserRouter>
        <PopularCoinItem coin={coin} />
      </BrowserRouter>,
    )
    const img = screen.queryByRole('img') as HTMLImageElement
    expect(img.src).toContain(
      'https://assets.coincap.io/assets/icons/btc@2x.png',
    )
  })

  test('Onclick must call navigate to /coin/bitcoin', () => {
    render(
      <BrowserRouter>
        <PopularCoinItem coin={coin} />
      </BrowserRouter>,
    )
    const coinItem = screen.getByTestId('coin-item')
    userEvent.click(coinItem)
    expect(window.location.href).toBe('http://localhost/coin/bitcoin')
  })
})
