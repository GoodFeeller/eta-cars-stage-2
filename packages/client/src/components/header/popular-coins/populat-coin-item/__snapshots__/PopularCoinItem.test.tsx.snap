// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularCoinItem test PopularCoinItem snapshot test 1`] = `
<div
  class="coinItem"
  data-testid="coin-item"
>
  <img
    alt="Not found"
    src="https://assets.coincap.io/assets/icons/btc@2x.png"
  />
  <span>
    BTC
  </span>
  <span>
    $21,000.23
  </span>
</div>
`;
